### ISSUE TITLE
Raven progress file

### ISSUE THREAD

**Issue by huard** [Ouranosinc/raven#44]
## Description

An option in Raven (and Ostrich) to create an output file storing the progress % from 0 to 100 and the estimated number of seconds until the process completion. If these could be saved in a json file, e.g. something like: 

```json
{
	"% progress": 65,
	"seconds remaining": 123
}
```
It doesn't have to be super accurate, especially for short jobs, just give a rough idea. 

On our end, we'd be able to read this file once in a while during execution to tell the user how far along is the simulation.

**Comment by julemai** [comment]
@huard 

re RAVEN: I will add an *.rvi file option `:PavicsMode` which will create that file (`Raven_progress.txt`) next to the `Raven_errors.txt`. We can add more PAVICS specific features under the `:PavicsMode` option later if needed.

re OSTRICH: I will reach out to Shawn and ask if he can do that with Ostrich. And offer that I do that in case he does not have time. But I'm not on his SVN etc... So better check before.

**Comment by julemai** [comment]
@huard 

re RAVEN: The progress file is now written as soon as you specify `:PavicsMode` in the *.rvi file. It will update the `Raven_progress.txt` file at each simulated time step and contains the above mentioned JSON string. 

Be aware that the estimates don't include I/O of any kind. it is just simulation time. The I/O highly depends on the model size and the amount of output you want it to write. hence it is hard to extrapolate the required time. For lumped models it should be pretty precise though.

I have uploaded the new RAVEN code rev156 (<a href="https://github.com/Ouranosinc/raven/wiki/raven-template-setups/Raven_rev156.zip" download="Raven_rev156.zip">.zip</a>). It is also added to the documentation ([here](https://github.com/Ouranosinc/raven/wiki/Technical-Notes)) right below the Table of Contents. 

I've also resolve the CEMA_NIEGE typo in RAVEN. It supports now both spellings- SNOBAL_CEMA_NEIGE and SNOBAL_CEMA_NIEGE to be backward compatible.

Working now on Ostrich. Will come soon...

**Comment by julemai** [comment]
@huard 

re OSTRICH: The progress file `OstProgress0.txt` is always written- no need of a specific setting in the `ostIn.txt` setup file. 

I have uploaded the new OSTRICH code v2017-12-19+ (<a href="https://github.com/Ouranosinc/raven/wiki/raven-template-setups/Ostrich_2017-12-19_plus_progressJSON.zip" download="Ostrich_2017-12-19_plus_progressJSON.zip">.zip</a>). It is also added to the documentation (here) right below the Table of Contents.

I have reported the changes to Shawn Matott. He will include them in the next official release. He also told me that he is working on moving towards GitHub with his source code.
